docs/odata/doAndDont.markdown

```diff
@@ -0,0 +1,109 @@
+---
+layout: default
+title: Dos and Don'ts
+name: doAndDont
+parent: oData
+nav_order: 4
+---
+
+## Dos and Don'ts
+
+This section contains useful dos and don'ts for developers programming SAP Gateway applications and optimizing performance.
+
+### Dos
+
+#### * Do think in REST terms
+
+If you find yourself inventing function imports such as CreateX, ReadX, GetX, ChangeX, UpdateX, or DeleteX, you should make the entity type X part of your model.
+`CRUD` is covered by `GET`, `PUT`, and `DELETE` to entities and `POST` to entity sets. .
+
+Function imports may be used for operations other than `CRUD`, such as Approve, Reject, Reserve, and Cancel. However, you might find that a reservation entity that you can create, update, and delete is better suited.
+
+#### * Do return completely filled entries in response to GET ~/EntitySet
+
+Think:
+
+```sql
+SELECT * FROM EntitySet WHERE ($filter) ORDER BY ($orderby) UP TO ($top) ROWS
+```
+
+if you see the following:
+
+```sql
+GET ~/EntitySet?$filter=...&$orderby=...&$top=...
+```
+
+OData does not distinguish GetList from GetDetail. OData clients expect to obtain all information from reading an entity set. If the information is not transparent and comprehensive, OData clients might attempt subsequent updates that could lead to errors in your data.
```

Comment:
this is a little unclear to me - are you referring to GETting an entityset vs a single entity?

Reply:
ah, now i understand your rationale. but the initial GET of an entity set, including its' binding to a UI5 control, then re-using data/binding for a relative binding upon "nav to detail" is UI5-specific, not an OData feature per se.
so i think rewording the paragraph would make this clearer.